app/workers/audio_callback_worker.rb

```diff
@@ -9,6 +9,17 @@ class UnknownAudioTypeError < StandardError; end
 
   def perform(_sqs_msg, job)
     audio = AudioFile.find(job['id'])
+    update_audio(audio, job)
```

Comment:
I was also playing with putting this update method inside the lock block as well - I may still do that, but in any case, I find this more readable to separate the methods.

Reply:
Yeah, it's a bit of an edge case to get 2 audio callbacks at once.  Then again, this whole debacle is about edge cases.